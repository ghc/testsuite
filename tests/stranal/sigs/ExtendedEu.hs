{-# LANGUAGE BangPatterns #-}

module ExtendedEu where

extendedEu :: Int -> Int -> (Int, Int)
extendedEu a 0 = (1, 0)
extendedEu a b = let b' = s - q * t
                 in b' `seq` (t, b')
    where (q, r) = quotRem a b
          (s, t) = extendedEu b r
