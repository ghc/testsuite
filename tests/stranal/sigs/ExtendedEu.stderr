
==================== Strictness signatures ====================
ExtendedEu.extendedEu: <L,1*U(U)><S(S),1*U(U)>dm(tm(d),tm(d))


